### PR TITLE
[CAS-1504] Allow customizing memberLimit when querying channels

### PR DESCRIPTION
**Pull request by adasiewiczr**
https://stream-io.atlassian.net/browse/CAS-1504

### 🎯 Goal
Allow customizing `memberLimit` when querying channels

### 🛠 Implementation details

Added `memberLimit` parameter to `ChannelListViewModel`, `ChannelListViewModelFactory`, `ChatDomain::queryChannels` and `ChatDomain:: queryChannelsLoadMore `

### 🧪 Testing

Check if query channels and load more work fine.
Experiment with newly added parameter

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

**Review by amitkma — APPROVED** [review]
Looks good to me

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/factory/ChannelListViewModelFactory.kt [hunk 1]
```diff
@@ -17,8 +17,8 @@ import io.getstream.chat.android.ui.channel.list.viewmodel.ChannelListViewModel
  * @param filter How to filter the channels.
  * @param sort How to sort the channels, defaults to last_updated.
  * @param limit How many channels to return.
- * @param messageLimit The number of messages to fetch for each channel.
- * @param chatEventHandler The instance of [ChatEventHandler] that will be used to handle channel updates event for this combination of [sort] and [filter].
+ * @param memberLimit The number of members per channel.
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/factory/ChannelListViewModelFactory.kt:20]: * @param messageLimit The number of messages to fetch for each channel.

**Review by bychkovdmitry — APPROVED** [review]
Great, just one minor comment!

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/factory/ChannelListViewModelFactory.kt (hunk 1 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/factory/ChannelListViewModelFactory.kt:20]: nice catch!

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt [hunk 2]
```diff
@@ -125,10 +125,15 @@ public class ChannelListViewModel(
...
     /**
-     * Default value of the number of messages to include in each channel when querying channels,
+     * Default value of the number of messages to include in each channel when querying channels.
      */
     private val MESSAGE_LIMIT = 1
 
+    /**
+     * Default value of the number of members to include in each channel when querying channels.
+     */
+    private val MEMBER_LIMIT = 30
```
Comment (filbabic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:135]: Could we push these as const? I think I mentioned this somewhere before but it got neglected (not on this PR). 

You can put it at the bottom of the file outside the ViewModel and then make it a `const`.

Cc @bychkovdmitry if we want to put it at the bottom or top.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt [hunk 3]
```diff
@@ -43,14 +43,16 @@ import kotlinx.coroutines.launch
...
  * @param messageLimit The number of messages to fetch for each channel.
- * @param chatEventHandler The instance of [ChatEventHandler] that will be used to handle channel updates event for this combination of [sort] and [filter].
+ * @param memberLimit The number of members to fetch per channel.
+ * @param chatEventHandlerFactory The instance of [ChatEventHandlerFactory] that will be used to create [ChatEventHandler].
  */
 public class ChannelListViewModel(
     private val chatDomain: ChatDomain = ChatDomain.instance(),
     private val filter: FilterObject? = null,
     private val sort: QuerySort<Channel> = DEFAULT_SORT,
     private val limit: Int = 30,
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt:53]: Same here - these would be great as a constant!

**Review by filbabic — APPROVED** [review]
Looks good - I left a couple of comments about constants, but the rest is fine!

Since the comments are part Compose, I'll pull it out as a separate compose ticket and approve it for now.

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**